### DISCUSSION TITLE
Slingbox connection stopped working around Midnight

### DISCUSSION THREAD

**Discussion by knoppix1337** [GerryDazoo/Slinger#280] [General]
Version : 3.08d Running on Linux-5.15.76-v7+-armv7l-with-glibc2.31 pid= 671
Using config file config.ini
Connection Manager Running on port 8080 with 10 max streams using URL slingbox.
Streamer Running:  starting up on  port 808010  8388608config.ini
 SLINGBOX 8080 8080 8388608
8080 No valid slingbox ip info found in config.ini
Finding Slingbox on local network. My IP Info =  b8:27:eb:75:b4:a4
Finding Slingbox on local network. My IP Info =  192.168.1.16
Slingbox Found 192.168.1.11 5001 " SA_Slingbox " FinderID 
Reading Custom Remote definition from Tivo.Series2-3.Remote.txt
 * Serving Flask app 'slingbox_server'
 * Debug mode: off
8080 Using slingbox at  ('192.168.1.11', 5001)
Streamer:  8080 Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
11/10/2022, 00:17:00.033   Streaming request from ('192.168.1.12', 39572)

Slinginfo  "Solo/Pro/ProHD" 10 30 192.168.1.11 5001 3000 10
8080 Starting Stream for  192.168.1.12:39572
Connecting... ('192.168.1.11', 5001) Control
cmd: 0x67 err: 0x2b 8
8080 Error Starting Session. Check your admin password in config.ini file!
8080 ERROR: Slingbox session startup failed.
8080 Logging Out
EXCEPTION [Errno 32] Broken pipe 0x101 0 0x68 0 3 0 0x2000 0x0
11/10/2022, 00:31:49.600   Streaming request from ('192.168.1.13', 51264)
11/10/2022, 00:34:16.004   Streaming request from ('192.168.1.13', 51267)

Thanks for this project. I hope this isn't a fatal problem related to SB servers going down.

I tried rebooting the pi server box and the slingbox. Tried connecting from different machines.

Let me know if you need any other info.

**Comment by lepmail** [comment]
I am getting exactly the same "password" error on my ProHD.  Everything was working fine when I had last tried it yesterday.

**Comment by bradut** [comment]
I am also getting a similar error after the SlingBox server was shut down:
(My device is a PRO-HD)

Version : 3.08d Running on Windows-10-10.0.19044-SP0 pid= 25524
Using config file config.ini
Connection Manager Running on port 65444 with 10 max streams using URL slingbox.
Streamer Running:  10 config.ini SLINGBOX 65444 65444 8388608
starting up on  port 65444 8388608
Checking for slingbox at ('12.34.56.78', 5001)
('12.34.56.78', 5001) OK
Bradut TV Using slingbox at  ('12.34.56.78', 5001)
Streamer:  Bradut TV Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
Reading Custom Remote definition from remote.txt
 * Serving Flask app 'slingbox_server' (lazy loading)
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: off
11/10/2022, 00:01:55.490  Registering Slingboxes {'4A62CBE24FB68440B6024FA97C26FBB7': 65444}
Getting external IP address
Redirector, my external IP address 12.34.56.78
Registering 4A62CBE24FB68440B6024FA97C26FBB7:65444

Register Successful
11/10/2022, 00:02:18.234   Streaming request from ('12.34.56.78', 1464)
Starting remote stream 1

Slinginfo  "Solo/Pro/ProHD" 16 30 12.34.56.78 5001 3000 10
Bradut TV Starting Stream for  12.34.56.78:1464
Connecting... ('12.34.56.78', 5001) Control
cmd: 0x67 err: 0x34 8
Bradut TV Error Starting Session. Check your admin password in config.ini file!
Bradut TV ERROR: Slingbox session startup failed.
Bradut TV Logging Out
EXCEPTION [WinError 10054] An existing connection was forcibly closed by the remote host 0x101 0 0x68 0 3 0 0x2000 0x0
Exception in thread Thread-2 (streamer):
Traceback (most recent call last):
  File "slingbox_server.py", line 372, in sling_cmd
    s_ctl.sendall( cmd )
ConnectionResetError: [WinError 10054] An existing connection was forcibly closed by the remote host

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "threading.py", line 1016, in _bootstrap_inner
  File "threading.py", line 953, in run
  File "slingbox_server.py", line 964, in streamer
    if s_ctl : s_ctl = closecontrol(s_ctl)
  File "slingbox_server.py", line 576, in closecontrol
    if Solo: sling_cmd( 0x68, b'')
  File "slingbox_server.py", line 375, in sling_cmd
    exit(1)
NameError: name 'exit' is not defined

**Comment by bradut** [comment]
Maybe this helps with the investigation: 
Just got news from overseas where one of my HD-PRO SlingBoxes could be accessed by the SlingPlayer directly, without signing in, because it was on the same LAN as the computer. 
I thought that this arrangement will not be affected by the server shutdown, but the news is that it no longer works. 
Maybe the SlingBox calls home when accessed?

**Comment by GerryDazoo** [comment]
This is bad. My 500 and M1 have survived but I'm getting the same issue on my Solo.
Comment out line  576:  if Solo: sling_cmd( 0x68, b'')
to stop the second exception and having the thread hosed.
I'm investigating and also a little surprised we never encountered this in our pre-shutdown test.

**Reply by smt171** [reply]
So your 500 and M1 are still working and abrinf a picture w the Slinger setup? 

Did you unplug them prior to Nov 9? Or leave plugged in? 

Did you block the Slingbox servers ports just in case? I'd like to get a listing of those IPs to block on mine before I plug back in--if you possibly have from your testing.

**Reply by smt171** [reply]
Also should we all maybe do a firmware backup just in case? (Not that I know how yet exactly but someone might)

The 500 has a USB port. I don't see one on the M1. Both have ethernet as you know.

**Reply by jrbski** [reply]
I have 2 350's and an M2, and they both survived the 11/9 shutdown, but my solo is having the same password issue... I didn't block anything, or disconnect it. I do have a few devices that were not connected, 2 solos, and a pro-HD that I may be able to test in an 'off-line' mode when I get a chance.

**Comment by Mike7766** [comment]
Is there step by step instructions on how to do the slinger software

**Comment by Mike7766** [comment]
i need help please on set up

**Comment by simonirwin1** [comment]
> Maybe this helps with the investigation: Just got news from overseas where one of my HD-PRO SlingBoxes could be accessed by the SlingPlayer directly, without signing in, because it was on the same LAN as the computer. I thought that this arrangement will not be affected by the server shutdown, but the news is that it no longer works. Maybe the SlingBox calls home when accessed?

I have found this too and agree with the "phone home" theory. I have been using the Slingbox player v1 instead of 2 as it doesn't log in to Slingbox servers. I thought this would make me immune to the shutdown, but alas even v1 doesn't work now, so I think the slingbox tries to communicate with the servers even if you don't use the login process provided by Sling.

**Comment by GerryDazoo** [comment]
This is not an issue with the Slinger software. Migrating to discussions

**Comment by smt171** [comment]
Did you all have your Slingboxes plugged in when Nov 9th hit? Maybe Dish (the owner of Slingbox now) bricked then?

I unplugged mine somewhere around Nov 4th just in case. Haven't plugged back in yet because I didnt have the pi set up yet.
